x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/source_logic.ts

```diff
@@ -69,7 +84,19 @@ export const SourceLogic = kea<MakeLogicType<SourceValues, SourceActions>>({
     initializeSource: (sourceId: string) => ({ sourceId }),
     initializeFederatedSummary: (sourceId: string) => ({ sourceId }),
     searchContentSourceDocuments: (sourceId: string) => ({ sourceId }),
-    updateContentSource: (sourceId: string, source: { name: string }) => ({ sourceId, source }),
+    updateContentSource: (
+      sourceId: string,
+      source: {
+        name?: string;
+        indexing?: {
+          enabled?: boolean;
+          features?: {
+            thumbnails?: { enabled: boolean };
+            content_extraction?: { enabled: boolean };
+          };
+        };
+      }
+    ) => ({ sourceId, source }),
```

Comment:
Can we pull this out into an interface? Perhaps it already exists somewhere?

```JavaScript
updateContentSource: (sourceId: string, source: SomeInterface) => ({ sourceId, source }),
```

Reply:
Done 👍